logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/OfferStrategy.java

```diff
@@ -62,76 +64,71 @@ public OfferResult evaluate(ClusterState clusterState, Protos.Offer offer) {
      * Offer result
      */
     public static class OfferResult {
-        final boolean acceptable;
-        final Optional<String> reason;
+        final List<String> complaints;
```

Comment:
A list of complaints doesn't make sense. If a machine already runs the executor, it will fail on (up to) all rules, when in reality it's only "failing" one rule. Complaints should be prioritised.